crates/bitwarden-vault/src/cipher/cipher.rs

```diff
@@ -502,7 +512,20 @@ impl KeyDecryptable<SymmetricCryptoKey, CipherListView> for Cipher {
             collection_ids: self.collection_ids.clone(),
             name: self.name.decrypt_with_key(key).ok().unwrap_or_default(),
             sub_title: self.get_decrypted_subtitle(key).ok().unwrap_or_default(),
-            r#type: self.r#type,
+            r#type: match self.r#type {
+                CipherType::Login => {
+                    let login = self
+                        .login
+                        .as_ref()
+                        .ok_or(CryptoError::MissingField("login"))?;
```

Comment:
Should we move the `require!` macro to crypto? It's doing mostly the same thing as this.

Reply:
Maybe. But long term we should ideally not use this pattern as Cipher would use an enum directly and not allow Type login and missing login.